tests/suites/deploy/deploy_revision.sh

```diff
@@ -79,7 +79,27 @@ run_deploy_revision_refresh() {
 	juju deploy juju-qa-test --revision 23 --channel latest/edge
 	wait_for "juju-qa-test" "$(charm_rev "juju-qa-test" 23)"
 
-	juju refresh juju-qa-test
+	# NOTE:
+	# The following loop is specific to juju 3.0+ due to
+	# async charm download and should NOT be removed in
+	# a merge from 2.9.
+	attempt=0
```

Comment:
The run_deploy_revision_refresh test originally took advantage of deploy requiring a charm download in 2.9 to go faster. With async charm download that is no longer possible. An alternative to the loop is 
```
wait_for "juju-qa-test" "$(idle_condition "juju-qa-test")"
```